smart_open/tests/test_http.py

```diff
@@ -107,3 +108,24 @@ def test_headers(self):
         responses.add_callback(responses.GET, URL, callback=request_callback)
         reader = smart_open.http.open(URL, 'rb', headers={'Foo': 'bar'})
         self.assertEqual(reader.headers['Foo'], 'bar')
+
+    @responses.activate
+    def test_https_seek_forward(self):
+        """Did the seek forward over HTTPS work?"""
+        responses.add_callback(responses.GET, "https://localhost", callback=request_callback)
+
+        with smart_open.open("https://localhost", "rb") as fin:
+            fin.seek(10)
+            read_bytes = fin.read(size=10)
+            self.assertEqual(BYTES[10:20], read_bytes)
+
+    @responses.activate
+    def test_https_seek_revert(self):
+        """Did the seek revert over HTTPS work?"""
+        responses.add_callback(responses.GET, URLS, callback=request_callback)
+
+        with smart_open.open(URLS, "rb") as fin:
+            read_bytes_1 = fin.read(size=10)
+            fin.seek(-10, whence=smart_open.s3.CURRENT)
```

Comment:
This is good. Add another test case with fin.seek(0) (seek to start of file) as well, because that's also a common use case.

Reply:
Ok
https://github.com/interpolatio/smart_open/commit/20179e5bba9d18bc6e4dd337c7382f48b9ac8ac9